app/lib/collator_mint.py

```diff
@@ -93,11 +96,46 @@ def register_mint_collator(node_name, ss58_format, rotate_key=False):
         return None
 
 
+def collator_set_keys(node_name, para_id):
+    try:
+        # 1. Get collator account keypair
+        collator_keypair = get_derived_collator_keypair(node_name, get_network_ss58_format())
+        collator_account_address = collator_keypair.ss58_address
+        # 2. Check funds
+        log.info("Check that {} has enough funds".format(collator_account_address))
+        node_client = get_node_client(node_name)
+        collator_account_funds = get_account_funds(node_client.url, collator_account_address)
+        if collator_account_funds < 0.5 * 10 ** 9:
+            # 3. Add funds with teleport
+            log.info(f"Funding {collator_account_address} (funds={collator_account_funds}) via Teleport from relay-chain")
+            relay_chain_client = get_relay_chain_client()
+            sudo_keypair =  Keypair.create_from_seed(network_sudo_seed())
+            teleport_result = teleport_funds(relay_chain_client, sudo_keypair, para_id, [collator_account_address], 1 * 10 ** 12)
+            if not teleport_result:
+                # exit, failed to fund account
+                log.error("Unable fund account: {}, node: {}".format(
+                    collator_account_address, getattr(relay_chain_client, 'url', 'NO_URL')))
+                return None
+            # 4. Check node has aura key
+            aura_public_key = "0x" + collator_keypair.public_key.hex()
+            if not node_keystore_has_key(node_client, 'aura', aura_public_key):
+                log.error(f'Node ({node_name}) doesn\'t have the required aura key in its keystore')
+                return None
+            # 5. Setting aura key on chain via "set session key"
+            set_session_key_result = set_node_session_key(node_client.url, collator_keypair.seed_hex, aura_public_key)
+            if not set_session_key_result:
+                # exit, failed to rotate key
+                log.error("Unable to rotate session key for node: {}".format(getattr(node_client, 'url', 'NO_URL')))
+                return None
```

Comment:
this part of code will be executed only on first call, the next  will skip it because of previous if condition `if collator_account_funds < 0.5 * 10 ** 9:`.

Example: 
1. The first call fails because of aura not in keystore
2. The next call will never reach same code since balance is not changed.

Reply:
Good remark, I fixed the condition.